src/rafimage.cpp

```diff
@@ -296,6 +292,37 @@ namespace Exiv2 {
         io_->read(jpg_img_length, 4);
         long jpg_img_off = Exiv2::getULong((const byte *) jpg_img_offset, bigEndian);
         long jpg_img_len = Exiv2::getULong((const byte *) jpg_img_length, bigEndian);
+        byte cfa_header_offset [4];
+        io_->read(cfa_header_offset, 4);
+        // byte cfa_header_length [4];
+        // io_->read(cfa_header_length, 4);
+        long cfa_hdr_off = Exiv2::getULong((const byte *) cfa_header_offset, bigEndian);
+        // long cfa_hdr_len = Exiv2::getULong((const byte *) cfa_header_length, bigEndian);
+        io_->seek(cfa_hdr_off,BasicIo::beg);
+        byte byte_count[4];
+        io_->read(byte_count, 4);
+        byte byte_tag[2];
+        byte byte_size[2];
+        int32_t count = getLong(byte_count, bigEndian);
```

Comment:
Please add a sanity check for this value, otherwise this will create a near-infinite loop.

Reply:
What is a sane value for you? The used test file has 32 tags, but I have no clue if other RAF files have a different number of raf tags.

Reply:
If there is no upper limit mandated by a standart, then you can make a rough check like: `enforce(count < remaining_file_size/size_of_a_tag);`, provided that this is at least an order of magnitude less than the maximum value of `int32_t` for all reasonable files.

If I understand your code, then aren't you searching for the pixelwidth and pixelheight inside the `cfa_hdr`? And you've just extracted its length a few lines above, so given the tag size and the length, that gives you a pretty good upper limit on the number of tags that can possibly exist in there.

Reply:
tag size is not constant according to exiftool, but I guess I could add a check with the minimal size for at least something